nemo/collections/asr/parts/utils/diarization_utils.py

```diff
@@ -15,498 +15,203 @@
-                logits_len = torch.from_numpy(np.array([len(greedy_predictions_list)]))
-                greedy_predictions_list = greedy_predictions_list[onset_delay:-mid_delay]
-                greedy_predictions = torch.from_numpy(np.array(greedy_predictions_list)).unsqueeze(0)
-                text, char_ts, word_ts = wer_ts.ctc_decoder_predictions_tensor_with_ts(
-                    self.model_stride_in_secs, greedy_predictions, predictions_len=logits_len
-                )
-                _trans_words, word_ts = text[0].split(), word_ts[0]
-                words_list.append(_trans_words)
-                word_ts_list.append(word_ts)
-                assert len(_trans_words) == len(word_ts)
-        return words_list, word_ts_list
-
-    def apply_delay(self, word_ts):
-        """
-        Compensate the constant delay in the decoder output.
-
-        Arg:
-            word_ts (list): List contains the timestamps of the word sequences.
-
-        Return:
-            word_ts (list): List of the delay-applied word-timestamp values.
-        """
-        for p in range(len(word_ts)):
-            word_ts[p] = [
-                max(round(word_ts[p][0] - self.asr_delay_sec, 2), 0),
-                round(word_ts[p][1] - self.asr_delay_sec, 2),
-            ]
-        return word_ts
-
-    def save_VAD_labels_list(self, word_ts_list, audio_file_list):
+    def save_VAD_labels_list(self, word_ts_dict):
         """
-        Get non_speech labels from logit output. The logit output is obtained from
+        Takes the non_speech labels from logit output. The logit output is obtained from
         run_ASR() function.
 
         Args:
-            word_ts_list (list):
-                List that contains word timestamps.
+            word_ts_dict (dict):
+                A list containing word timestamps.
             audio_file_list (list):
-                List of audio file paths.
+                A list of audio file paths.
         """
         self.VAD_RTTM_MAP = {}
-        for i, word_timestamps in enumerate(word_ts_list):
-            speech_labels_float = self._get_speech_labels_from_decoded_prediction(word_timestamps)
+        for idx, (uniq_id, word_timestamps) in enumerate(word_ts_dict.items()):
+            speech_labels_float = self.get_speech_labels_from_decoded_prediction(word_timestamps)
             speech_labels = self.get_str_speech_labels(speech_labels_float)
-            uniq_id = get_uniqname_from_filepath(audio_file_list[i])
             output_path = os.path.join(self.root_path, 'pred_rttms')
             if not os.path.exists(output_path):
                 os.makedirs(output_path)
             filename = labels_to_rttmfile(speech_labels, uniq_id, output_path)
-            self.VAD_RTTM_MAP[uniq_id] = {'audio_filepath': audio_file_list[i], 'rttm_filepath': filename}
+            self.VAD_RTTM_MAP[uniq_id] = {'audio_filepath': self.audio_file_list[idx], 'rttm_filepath': filename}
```

Comment:
you can remove reference by idx, by getting audio_file_name using AUDIO_RTTM_MAP

Reply:
Its always better to get filenames using uniq_id key than idexes